tests/Parsing/DanglingCommentTest.elm

```diff
@@ -0,0 +1,353 @@
+
+                Nothing ->
+                    Expect.fail "Failed to parse the module!"
+    in
+    Test.describe
+        "Test the parsing function for dangling comments."
+        [ Test.describe "Modules with no dangling comments."
+            (List.indexedMap
+                (\idx ->
+                    \moduleStr ->
+                        Test.test
+                            ("no dangling comments " ++ toString idx)
+                            (\() -> checkNoDanglingComments moduleStr)
+                )
+                [ emptyModuleWithDoc
+                , emptyModuleWithoutDoc
+                , moduleNoImportsWithDoc
+                , moduleNoImportsWithoutDoc
+                , moduleNoDeclarationsWithDoc
+                , moduleNoDeclarationsWithoutDoc
+                , moduleWithDocumentedTypeAndDoc
+                , moduleWithDocumentedTypeAndNoDoc
+                , moduleWithDocumentedTypeAndImportsAndDoc
+                , largerExample
+                ]
+            )
+        , Test.test "Module with wrong top-level parsed as dangling (1)." <|
+            \() ->
+                checkExpectation
+                    (expectedComment 4
+                        0
+                        5
+                        2
+                        "{- This module is empty. One day, though...\n-}"
+                    )
+                    moduleWithWrongDocsType1
+        , Test.test "Module with wrong top-level parsed as dangling (2)." <|
+            \() ->
+                checkExpectation
+                    (expectedComment 4
+                        0
+                        4
+                        43
+                        "-- This module is empty. One day, though..."
+                    )
+                    moduleWithWrongDocsType2
+        , Test.test "Module with dangling comment 1." <|
+            \() ->
+                checkExpectation
+                    (expectedComment 11
+                        0
+                        11
+                        26
+                        "-- just a dangling comment"
+                    )
+                    moduleWithDanglingComment1
+        , Test.test "Module with dangling comment 2." <|
+            \() ->
+                checkExpectation
+                    (expectedComment 11
+                        0
+                        11
+                        29
+                        "{- just a dangling comment -}"
+                    )
+                    moduleWithDanglingComment2
+        , Test.test "Module with dangling comment 3." <|
+            \() ->
+                checkExpectation
+                    (expectedComment 11
+                        0
+                        11
+                        30
+                        "{-| just a dangling comment -}"
+                    )
+                    moduleWithDanglingComment3
+        ]
+
+
+emptyModuleWithDoc : String
```

Comment:
As far as I can see, you only use these strings once -- so it would make sense for me to include to in-line them? And if you use them multiple-times, I'd still write them out to make the tests more readable through redundancy.

Otherwise, it's nigh impossible to spot a bug in the test when the actual string is referred and defined elsewhere. Try to sympathize with the reader rather than the writer / aesthetics of the source file -- if you didn't write the test and need to understand it / check it for correctness, would it be easier for you to jump around the file or to read the strings immediately in the test?

Reply:
Good point :)